internal/cmd/gocdk/init_test.go

```diff
@@ -147,6 +148,83 @@ func TestInit(t *testing.T) {
 		}
 	})
 }
+func TestInferModulePath(t *testing.T) {
+	t.Run("NoGOPATHEntry", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", testTempDirPrefix)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			if err := os.RemoveAll(dir); err != nil {
+				t.Error(err)
+			}
+		}()
+		ctx := context.Background()
+		pctx := &processContext{
+			workdir: dir,
+			env:     []string{},
+			stdin:   strings.NewReader(""),
+			stdout:  ioutil.Discard,
+			stderr:  ioutil.Discard,
+		}
+
+		fmt.Println(pctx.env)
+
+		const projectName = "myspecialproject"
+		if err := run(ctx, pctx, []string{"init", projectName}); err == nil {
+			t.Errorf("inferModulePath should error when --module-path is not" +
+				" supplied and GOPATH is empty")
+		}
+	})
+	t.Run("SingleGOPATHEntry", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", testTempDirPrefix)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			if err := os.RemoveAll(dir); err != nil {
+				t.Error(err)
+			}
+		}()
+		ctx := context.Background()
+		pctx := &processContext{
+			workdir: dir,
+			env:     []string{"GOPATH=" + dir},
+			stdin:   strings.NewReader(""),
+			stdout:  ioutil.Discard,
+			stderr:  ioutil.Discard,
+		}
+		fmt.Println(pctx.env)
+		const projectName = "myspecialproject"
+		if err := run(ctx, pctx, []string{"init", projectName}); err != nil {
+			t.Errorf("inferModulePath errored: %+v", err)
+		}
+	})
+	t.Run("MultipleGOPATHEntries", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", testTempDirPrefix)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			if err := os.RemoveAll(dir); err != nil {
+				t.Error(err)
+			}
+		}()
+		ctx := context.Background()
+		pctx := &processContext{
+			workdir: dir,
+			env:     []string{"GOPATH=" + dir + ":~/some/other/path"},
```

Comment:
I'm not sure whether it's significant for this test, but `~/some/other/path` isn't a valid path for Windows (different path separators) or Unix (tilde expansion occurs at the shell, not in path resolution).